lxd/instance/drivers/driver_qemu.go

```diff
@@ -7426,22 +7432,91 @@ func (d *qemu) DeviceEventHandler(runConf *deviceConfig.RunConfig) error {
 	return nil
 }
 
-// vsockID returns the vsock Context ID for the VM.
-func (d *qemu) vsockID() int {
-	// We use the system's own VsockID as the base.
-	//
-	// This is either "2" for a physical system or the VM's own id if
-	// running inside of a VM.
-	//
-	// To this we add 1 for backward compatibility with prior logic
-	// which would start at id 3 rather than id 2. Removing that offset
-	// would cause conflicts between existing VMs until they're all rebooted.
-	//
-	// We then add the VM's own instance id (1 or higher) to give us a
-	// unique, non-clashing context ID for our guest.
+// getVsockID returns the vsock Context ID for the VM.
+// If no error is returned, a vsock Context ID of 0 indicates the value is not set for the VM.
```

Comment:
Lets just have this function return an error if not set. Saves having every caller check for that scenario.

Reply:
Changed.